atom/browser/native_window_mac.mm

```diff
@@ -1127,15 +1127,20 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
                                      const NativeWindow::ProgressState state) {
   NSDockTile* dock_tile = [NSApp dockTile];
 
+  // Sometimes macOS would install a default contentView for dock, we must
+  // verify whether NSProgressIndicator has been installed.
+  bool first_time = !dock_tile.contentView ||
+                    [[dock_tile.contentView subviews] count] == 0 ||
+                    ![[[dock_tile.contentView subviews] lastObject]
+                        isKindOfClass:[NSProgressIndicator class]];
```

Comment:
Can we record this explicitly rather than using the state of the dock tile as an implicit state variable? e.g. `has_installed_indicator_` or something? & perhaps store a reference to the `NSProgressIndicator`?

Reply:
The `dockTile.contentView` is not guaranteed to keep unchanged once after setting, in my case simply refreshing the tests window would reset `dockTile.contentView` to some kind of default view.

So we could not effectively track whether the progress bar is installed in `dockTile.contentView`, as the change was made by macOS or Chromium or something else.

This only happens on very few machines, I doubt it is some bug of macOS.